lib/spack/llnl/util/filesystem.py

```diff
@@ -472,3 +474,110 @@ def find_library_path(libname, *paths):
         if os.path.exists(library):
             return path
     return None
+
+# Utilities for libraries
+
+
+class LibraryList(collections.Sequence):
+    """Manages a list of absolute paths to libraries"""
+    def __init__(self, libraries):
+        self.libraries = list(libraries)
+
+    @property
+    def directories(self):
+        return [os.path.dirname(x) for x in self.libraries]
+
+    @property
+    def libnames(self):
+        return [os.path.basename(x).split('.')[0] for x in self.libraries]
+
+    @property
+    def names(self):
```

Comment:
p.p.s. Now i see why you chose these names:

`libnames` is `lib`+`names` of libraries, whereas `names` are only names of the libraries.

I think that's the best one can do if you provide both properties accessible from class instantiations.